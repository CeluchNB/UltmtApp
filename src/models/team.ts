import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { LocalUser, Team } from '../types/team'

export const DisplayTeamSchema: Realm.ObjectSchema = {
    name: 'DisplayTeam',
    embedded: true,
    properties: {
        _id: 'string',
        place: 'string',
        name: 'string',
        teamname: 'string',
        seasonStart: 'string',
        seasonEnd: 'string',
    },
}

export const GuestTeamSchema: Realm.ObjectSchema = {
    name: 'GuestTeam',
    embedded: true,
    properties: {
        _id: 'string?',
        place: 'string?',
        name: 'string',
        teamname: 'string?',
        seasonStart: 'string?',
        seasonEnd: 'string?',
    },
}

export class TeamSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Team',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            place: 'string',
            name: 'string',
            teamname: 'string',
            managers: { type: 'list', objectType: 'DisplayUser' },
            players: { type: 'list', objectType: 'DisplayUser' },
            seasonStart: 'string',
            seasonEnd: 'string',
            seasonNumber: 'int',
            continuationId: 'string',
            rosterOpen: 'bool',
        },
    }

    _id: string
    place: string
    name: string
    teamname: string
    managers: DisplayUser[]
    players: LocalUser[]
    seasonStart: string
    seasonEnd: string
    seasonNumber: number
    continuationId: string
    rosterOpen: boolean

    constructor(team: Team) {
        this._id = team._id
        this.place = team.place
        this.name = team.name
        this.teamname = team.teamname
        this.managers = team.managers
        this.players = team.players
        this.seasonStart = team.seasonStart
        this.seasonEnd = team.seasonEnd
        this.seasonNumber = team.seasonNumber
        this.continuationId = team.continuationId
        this.rosterOpen = team.rosterOpen
    }
}
