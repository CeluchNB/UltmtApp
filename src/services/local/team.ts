import { Realm } from '@realm/react'
import { Team } from '../../types/team'
import { TeamSchema } from '../../models'
import { getRealm } from '../../models/realm'

export const saveTeams = async (teams: Team[]) => {
    const realm = await getRealm()
    realm.write(() => {
        for (const team of teams) {
            realm.create('Team', team, Realm.UpdateMode.Modified)
        }
    })
}

export const getManagingTeams = async (id: string): Promise<Team[]> => {
    const realm = await getRealm()
    const teams = await realm.objects<TeamSchema>('Team')
    return teams
        .filter(t => t.managers.map(m => m._id).includes(id))
        .map(t => parseTeam(t))
}

const parseTeam = (schema: TeamSchema): Team => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            place: schema.place,
            name: schema.name,
            teamname: schema.teamname,
            managers: schema.managers,
            players: schema.players,
            seasonStart: schema.seasonStart,
            seasonEnd: schema.seasonEnd,
            seasonNumber: schema.seasonNumber,
            continuationId: schema.continuationId,
            rosterOpen: schema.rosterOpen,
            requests: [],
            games: [],
        }),
    )
}
