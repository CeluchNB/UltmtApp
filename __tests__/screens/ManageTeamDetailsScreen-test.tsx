import * as ManagedTeamReducer from '../../src/store/reducers/features/team/managedTeamReducer'
import * as RequestData from '../../src/services/data/request'
import * as TeamData from '../../src/services/data/team'
import { DetailedRequest } from '../../src/types/request'
import ManageTeamDetailsScreen from '../../src/screens/ManageTeamDetailsScreen'
import { ManagedTeamDetailsProps } from '../../src/types/navigation'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { Team } from '../../src/types/team'
import { setTeam } from '../../src/store/reducers/features/team/managedTeamReducer'
import store from '../../src/store/store'
import { waitUntilRefreshComplete } from '../../fixtures/utils'
import { QueryClient, QueryClientProvider } from 'react-query'
import { act, fireEvent, render, screen } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const navigate = jest.fn()
const goBack = jest.fn()
const addListener = jest.fn().mockReturnValue(() => {})
const setOptions = jest.fn()

let getManagedTeamResponse: Team
let requestObject: DetailedRequest

const props: ManagedTeamDetailsProps = {
    navigation: {
        addListener,
        navigate,
        goBack,
        setOptions,
    } as any,
    route: {
        params: {
            id: 'id1',
            place: 'place',
            name: 'name',
        },
    } as any,
}

const client = new QueryClient({
    defaultOptions: {
        queries: { retry: 0 },
    },
})

beforeEach(() => {
    jest.clearAllMocks()
    getManagedTeamResponse = {
        _id: 'id1',
        place: 'place',
        name: 'name',
        teamname: 'placename',
        players: [
            {
                _id: 'playerid1',
                firstName: 'first1',
                lastName: 'last1',
                username: 'first1last1',
            },
        ],
        requests: ['request1', 'request2'],
        managers: [
            {
                _id: 'playerid4',
                firstName: 'first4',
                lastName: 'last4',
                username: 'first4last4',
            },
        ],
        seasonNumber: 1,
        seasonStart: '2022',
        seasonEnd: '2022',
        continuationId: 'id1234',
        rosterOpen: false,
        games: [],
    }
    requestObject = {
        _id: 'request1',
        team: 'id1',
        user: 'playerid1',
        requestSource: 'team',
        teamDetails: {
            _id: 'id1',
            place: 'place',
            name: 'name',
            teamname: 'placename',
            seasonStart: '2022',
            seasonEnd: '2022',
        },
        userDetails: {
            _id: 'playerid1',
            firstName: 'first1',
            lastName: 'last1',
            username: 'first1last1',
        },
        status: 'pending',
    }
    jest.spyOn(RequestData, 'getRequest').mockImplementation(
        async requestId => {
            if (requestId === 'request1') {
                return {
                    ...requestObject,
                    _id: 'request1',
                    requestSource: 'team',
                    userDetails: {
                        _id: 'playerid2',
                        firstName: 'first2',
                        lastName: 'last2',
                        username: 'first2last2',
                    },
                }
            } else if (requestId === 'request2') {
                return {
                    ...requestObject,
                    _id: 'request2',
                    requestSource: 'player',
                    userDetails: {
                        _id: 'playerid3',
                        firstName: 'first3',
                        lastName: 'last3',
                        username: 'first3last3',
                    },
                }
            }
            return requestObject
        },
    )
    jest.spyOn(TeamData, 'getManagedTeam').mockReturnValue(
        Promise.resolve(getManagedTeamResponse),
    )
    store.dispatch(setTeam(getManagedTeamResponse))
})

describe('ManageTeamDetailsScreen', () => {
    beforeAll(() => {
        jest.useFakeTimers()
    })

    afterAll(() => {
        jest.useRealTimers()
    })

    it('should match snapshot', async () => {
        const snapshot = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(snapshot.getByTestId('mtd-flat-list'))

        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('should navigate to rollover a team with pending requests', async () => {
        const { getByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const button = getByText('Rollover Team')

        fireEvent.press(button)

        expect(navigate).toHaveBeenCalledWith('RolloverTeam')
    })

    it('should handle swipe to refresh', async () => {
        const teamSpy = jest.spyOn(TeamData, 'getManagedTeam')

        const { getByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        // expect getManagedTeam to be called
        expect(teamSpy).toHaveBeenCalled()

        // check data is passed and displayed correctly
        const teamname = getByText('@placename')
        expect(teamname).toBeTruthy()

        // player displayed correctly
        const player = getByText('@first1last1')
        expect(player).toBeTruthy()
    })

    it('should handle a page load error', async () => {
        jest.spyOn(TeamData, 'getManagedTeam').mockReturnValueOnce(
            Promise.reject({ message: 'test error' }),
        )

        const { getByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const errorText = getByText('test error')
        expect(errorText).toBeTruthy()
    })

    it('should remove player correctly', async () => {
        const removeSpy = jest
            .spyOn(TeamData, 'removePlayer')
            .mockImplementationOnce(async () => {
                return { ...getManagedTeamResponse, players: [] }
            })

        const { queryByText, getAllByTestId, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const username = '@first1last1'
        expect(queryByText(username)).not.toBeNull()
        const deleteButtons = getAllByTestId('delete-button')
        fireEvent.press(deleteButtons[0])
        /*
        Should not need act() after fireEvent, however test fails if it's not there.
        The alternative, preferred method of

        fireEvent.press(acceptButtons[0])
        await waitFor(() => queryByText('@first3last3'))

        gives an error of:
        You called act(async () => ...) without await...

        Explore fix in the future
        Another option is wrapping fireEvent in act, but
        this makes it more obvious something weird is going on
    */
        await act(async () => {})

        expect(removeSpy).toHaveBeenCalled()
        expect(queryByText(username)).toBeNull()
    })

    it('should handle error swipe', async () => {
        jest.spyOn(TeamData, 'getManagedTeam').mockClear()
        const spy = jest
            .spyOn(TeamData, 'getManagedTeam')
            .mockReturnValue(
                Promise.reject({ message: 'test error getting team' }),
            )

        const { queryByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        expect(queryByText('test error getting team')).not.toBeNull()

        const scrollViewError = getByTestId('mtd-flat-list')
        const { refreshControl: refreshControlError } = scrollViewError.props
        await act(async () => {
            refreshControlError.props.onRefresh()
        })

        expect(spy).toHaveBeenCalled()
    })

    it('should navigate to public user', async () => {
        const { getByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const username = getByText('@first1last1')
        fireEvent.press(username)

        expect(navigate).toHaveBeenCalled()
    })

    it('should handle add players text', async () => {
        const { getAllByTestId, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const buttons = getAllByTestId('create-button')
        fireEvent.press(buttons[1])

        expect(navigate).toHaveBeenCalled()
    })

    it('should navigate to public user screen on manager click', async () => {
        const { getByText, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const button = getByText('@first4last4')
        fireEvent.press(button)

        expect(navigate).toHaveBeenCalled()
    })

    it('should navigate to request manager screen', async () => {
        const { getAllByTestId, getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        await waitUntilRefreshComplete(getByTestId('mtd-flat-list'))

        const buttons = getAllByTestId('create-button')
        fireEvent.press(buttons[0])

        expect(navigate).toHaveBeenCalled()
    })

    it('should toggle roster status', async () => {
        const dataFn = jest
            .fn()
            .mockImplementationOnce(async (_token, id, open) => {
                return { rosterOpen: open } as Team
            })
        const spy = jest.spyOn(ManagedTeamReducer, 'toggleRosterStatus')

        jest.spyOn(TeamData, 'toggleRosterStatus').mockImplementationOnce(
            dataFn,
        )
        render(
            <Provider store={store}>
                <NavigationContainer>
                    <QueryClientProvider client={client}>
                        <ManageTeamDetailsScreen {...props} />
                    </QueryClientProvider>
                </NavigationContainer>
            </Provider>,
        )

        const toggleSwitch = screen.getByTestId('roster-open-switch')
        fireEvent(toggleSwitch, 'valueChange')

        expect(spy).toHaveBeenCalled()
    })
})
