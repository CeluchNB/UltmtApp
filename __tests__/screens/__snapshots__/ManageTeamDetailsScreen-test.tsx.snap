// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle swipe to refresh 1`] = `null`;

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "flex": 1,
      "height": "100%",
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "#31adff",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    testID="mtd-flat-list"
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#ffd600",
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          place name
        </Text>
        <Text
          style={
            Object {
              "color": "#ffd600",
              "fontSize": 30.821917808219176,
              "fontWeight": "600",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          2022
        </Text>
        <Text
          style={
            Object {
              "color": "#ffd600",
              "fontSize": 30.821917808219176,
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          @
          placename
        </Text>
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#ffd600",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 2,
              "minWidth": "auto",
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      Object {
                        "marginHorizontal": 8,
                      },
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                manage requests
              </Text>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#ffd600",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "elevation": 0,
              "marginTop": 10,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Start New Season
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "75%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffd600",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Managers
            </Text>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ffd600",
                      "borderRadius": 5,
                      "display": "flex",
                      "flexWrap": "nowrap",
                    },
                  ],
                ]
              }
              testID="create-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#303030",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰐕
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  first4 last4
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffd600",
                      "fontSize": 30.821917808219176,
                      "fontWeight": "normal",
                    }
                  }
                >
                  @
                  first4last4
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ffd600",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffd600",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Players
            </Text>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "backgroundColor": "#ffd600",
                      "borderRadius": 5,
                      "display": "flex",
                      "flexWrap": "nowrap",
                    },
                  ],
                ]
              }
              testID="create-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#303030",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰐕
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  first1 last1
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffd600",
                      "fontSize": 30.821917808219176,
                      "fontWeight": "normal",
                    }
                  }
                >
                  @
                  first1last1
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="delete-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff0000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅖
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ffd600",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
